backend/src/main/java/com/gestionnaire_de_stage/model/PasswordResetToken.java

```diff
@@ -0,0 +1,48 @@
+package com.gestionnaire_de_stage.model;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Date;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+public class PasswordResetToken {
+    private static final SecureRandom secureRandom = new SecureRandom();
+    private static final Base64.Encoder base64Encoder = Base64.getUrlEncoder();
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+    @Column(unique = true)
+    private String token;
+    @ManyToOne
+    private User user;
+    @Temporal(TemporalType.TIMESTAMP)
+    @Column(nullable = false, updatable = false)
+    private Date created;
+    private boolean unusable;//TODO maybe un puller?
```

Comment:
```suggestion
    private boolean unusable
```
pas de puller, au moment du call on checkera si la date a dépassé l'interval de temps alloué . Si ça dépasse, on dit à l'utilisateur que le token a expiré et il doit refaire sa demande, sinon on le laise update son mdp

Reply:
C'est pour ca que c'est un todo